lib/user-methods.js

```diff
@@ -74,10 +76,35 @@ function normalizeSyncUrl(authUrl, syncUrl) {
     return new URL(syncUrl, baseUrl, false).toString();
 }
 
+// node-fetch supports setting a timeout as a nonstandard extension, but normal fetch doesn't
+function fetchWithTimeout(input, init) {
```

Comment:
Do we test this behaviour? I wonder if it would be much simpler (and less error-prone) to simply wrap the response in a [`Promise.race`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/race) with another promise that gets rejected after a timeout? Somewhat like https://stackoverflow.com/a/46946573/503899 is suggesting.

Reply:
The SO suggestion doesn't actually set the timeout for the request, meaning that it won't be cancelled until it hits some default timeout.

Reply:
`testRegisterServerOffline()` tests this code path. Actually canceling the request is important as it'll otherwise sit around forever.